### PR TITLE
Add installation options.

### PR DESCRIPTION
**Pull request by dikshagupta1**
Builder-HA: Enable users to install postgresql and minio separately

### CODE REVIEW

File: scripts/provision.sh [hunk 1]
```diff
@@ -272,23 +272,24 @@ EOT
 }
 
 install_frontend() {
-  if hab svc status habitat/builder-api >/dev/null 2>&1; then
+  #Check if api and datastore services are already configures or installed.
+  if sudo hab svc status habitat/builder-api >/dev/null 2>/dev/null; then
```
Comment (pozsgaic) [inline scripts/provision.sh:276]: We don't want to have sudo embedded in a script.  It is better to invoke sudo at the cmd line,

File: scripts/provision.sh [hunk 2]
```diff
@@ -329,6 +361,10 @@ options:
 
   --install-frontend    Provision a Frontend/API only.
 
+  --install-postgresql  Provision the datastore only.
+
+  --install-minio       Provision the minio server only.
+
   --generate-bootstrap   Generate a bootstrap to be used for scaling our API Frontends
```
Comment (mwrock) [inline scripts/provision.sh:368]: This isn't related to this change but while you are here, could you delete this line? We removed the `--generate-bootstrap` functionality but did not remove this doc note.

File: scripts/provision.sh [hunk 3]
```diff
@@ -311,6 +312,37 @@ install_frontend() {
...
+    echo "ERROR: --install-postgresql can not not be used if you are using"
+    echo "externally hosted PostgreSQL(RDS, Azure Database for PostgreSql etc)."
+    echo "Set PG_EXT_ENABLED=false to fix this error."
+    echo
+    return
+  fi
+
+  start_init
+  configure
+  start_datastore
```
Comment (mwrock) [inline scripts/provision.sh:327]: Similar to `start_builder` you will need to add:

```
    while [ ! -f /hab/svc/builder-datastore/config/pwfile ]
    do
      sleep 2
    done
    local pg_pass=$(cat /hab/svc/builder-datastore/config/pwfile)
    cat <<EOT > pg_pass.toml
[datastore]
password = "$pg_pass"
EOT
  hab config apply builder-api.default $(date +%s) pg_pass.toml
```
Probably want to extract that to its own function.

File: scripts/provision.sh [hunk 4]
```diff
@@ -311,6 +312,37 @@ install_frontend() {
...
+  if [ "${PG_EXT_ENABLED:-false}" = "true" ]; then
+    echo "ERROR: --install-postgresql can not not be used if you are using"
+    echo "externally hosted PostgreSQL(RDS, Azure Database for PostgreSql etc)."
+    echo "Set PG_EXT_ENABLED=false to fix this error."
+    echo
+    return
+  fi
+
+  start_init
+  configure
```
Comment (mwrock) [inline scripts/provision.sh:326]: need to call `init_datastore` before `start_datastore`

**Review by mwrock — CHANGES_REQUESTED** [review]

File: scripts/provision.sh (hunk 1 above)
Comment (pozsgaic) [reply scripts/provision.sh:276]: Ignore the above - we have been using sudo in this script and have it wrapped in a function.

**Review by mwrock — APPROVED** [review]

**Pull request merged**